python/ray/dag/compiled_dag_node.py

```diff
@@ -474,25 +494,31 @@ def exec_operation(
         self,
         class_handle,
         op_type: _DAGNodeOperationType,
+        overlap_gpu_communication: bool = False,
     ) -> bool:
         """
         An ExecutableTask corresponds to a DAGNode. It consists of three
         operations: READ, COMPUTE, and WRITE, which should be executed in
         order to ensure that each operation can read the correct intermediate
         result.
-
         Args:
             class_handle: The handle of the class to which the actor belongs.
             op_type: The type of the operation. Possible types are READ,
                 COMPUTE, and WRITE.
-
+            overlap_gpu_communication: Whether to overlap GPU communication with
+                computation during DAG execution to improve performance.
         Returns:
             True if the next operation should not be executed; otherwise, False.
         """
+        if self.output_type_hint.requires_nccl():
+            nccl_group_id = self.output_type_hint.nccl_group_id
+            ctx = ChannelContext.get_current()
+            ctx.set_current_stream(nccl_group_id, op_type.name)
```

Comment:
@stephanie-wang Rather than this code, is using python context manager what you had on mind, something like:

```
        if self.output_type_hint.requires_nccl():
            nccl_group_id = self.output_type_hint.nccl_group_id
            nccl_group = ChannelContext.get_current().nccl_groups[nccl_group_id]

        if op_type == _DAGNodeOperationType.READ:
            with nccl_group._recv_stream:
                return self._read()
        elif op_type == _DAGNodeOperationType.COMPUTE:
            with nccl_group._compute_stream:
                return self._compute(overlap_gpu_communication, class_handle)
        elif op_type == _DAGNodeOperationType.WRITE:
            with nccl_group._send_stream:
                return self._write()
```

Reply:
Yes, exactly.

Reply:
Changed to use python context manager